src/shared/json.c

```diff
@@ -1066,6 +1066,42 @@ double json_variant_real(JsonVariant *v) {
         return 0.0;
 }
 
+int json_variant_bytes(JsonVariant *v, uint8_t **p, size_t *n) {
+        assert_return(p, -EINVAL);
+        assert_return(n, -EINVAL);
+
+        if (!v) {
+                *n = 0;
+                return 0;
+        }
+        if (!json_variant_is_array(v))
+                return -EINVAL;
+
+        if (json_variant_elements(v) == 0) {
+                *n = 0;
+                return 0;
+        }
+
+        *p = new (uint8_t, json_variant_elements(v));
```

Comment:
Please drop space after `new`.

Also, please assign value to the arguments that store function result only when the function succeed, and prefix such arguments with `ret`:
```c
int json_variant_bytes(JsonVariant *v, uint8_t **ret, size_t *ret_size) {
        _cleanup_ uint8_t *buf = NULL;
        size_t n;

        assert(ret);
        assert(ret_size);

        if (!v)
                goto not_found;
        ...

        *ret = TAKE_PTR(buf);
        *ret_size = n;
        return 0;

not_found:
        *ret = NULL;
        *ret_size = 0;
        return 0;
}
```